### PR TITLE
Tempsensor

### PR DESCRIPTION
**Pull request by asadwaheed10**

### CODE REVIEW

File: tests/test_tempthingspeak.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import random
+import logging
+import time
+from unittest import TestCase, main
+from thingspeakwriter import ThingSpeakWriter
+from thingspeakreader import ThingSpeakReader
+from thingspeakinfo import (READ_KEY_D2, WRITE_KEY_D2, FEED_D2, GOOD_STATUS)
+
+class TestThingSpeak(TestCase):
```
Comment (marianarafaelwhite) [inline tests/test_tempthingspeak.py:9]: Should I rename the thingspeak test case that I have to be lightthingspeak?

File: tests/test_tempsensordb.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	def test_add_bad_record(self):
+		record = {'date': '2020-11-22',
+			  'time': '14:11:34',
+			  'fanStatus': 1}
+
+		self.__db.create_table()
+		error_msg = 'Record exists unexpectedly'
+		self.assertFalse(self.__db.record_exists(record), error_msg)
+
+#		self.assertRaises(Exception, self.__db.add_record, record)
```
Comment (marianarafaelwhite) [inline tests/test_tempsensordb.py]: Why is this commented out?

File: tests/test_tempsensordb.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+from unittest import TestCase, main, skipIf
+from tempDB import TempDB
+
+TEMP_DB = 'temp_tempsensor.db'
+TEMP_TABLE = 'temp_lightclapper'
+PREMADE_DB = 'tests/test_db/test_tempsensor.db'
```
Comment (marianarafaelwhite) [inline tests/test_tempsensordb.py:8]: Is this file included in the commit?

File: tempsensor/tempDB.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+SINGLE_RECORD = 1
+
+class SqliteDB(metaclass=abc.ABCMeta):
+	"""
+	DB to store data
+	"""
+
+	def __init__(self, db_file, name):
+		"""
+		Initializes TempDB Context Manager
```
Comment (marianarafaelwhite) [inline tempsensor/tempDB.py:19]: I think the comment here needs to be updated to the general name; however, I was thinking that maybe I could separate the SqliteDB into its own file in a common directory (maybe this weekend :) )

File: tempsensor/temp.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+		self._temp_poll_time = poll_time #Polling for the given time
+		self._data = dict()
+
+	def read_data(self):
+		self._data = bme280.sample(self._bus, self._address, self._calibration_params) #Taking in a single temperature reading
+		return self._data.temperature #Returning the reading
+
+def temperature_test():
+	sensor = Temperature()
+	while True:
```
Comment (marianarafaelwhite) [inline tempsensor/temp.py:23]: Just curious, is there any cleanup required for the temperature sensor?

**Review by marianarafaelwhite — APPROVED** [review]
Looks good to me! All my comments are minor and can come later as part of a final cleanup after this is merged to the main branch.

Also, I wanted to let you know that I use command line tools to auto format my python code that makes things easier. If you're interested, I use https://flake8.pycqa.org/en/latest/ to determine any syntax errors and https://pypi.org/project/autopep8/ to try to automatically fix any problems.

File: tempsensor/temp.py (hunk 5 above)
Comment (asadwaheed10) [reply tempsensor/temp.py:23]: From what I saw there wasn't, I can double check again before deploying to the Main branch this weekend

File: tests/test_tempsensordb.py (hunk 3 above)
Comment (asadwaheed10) [reply tests/test_tempsensordb.py:8]: They weren't, part of the final cleanup I can commit those files as well

File: tests/test_tempsensordb.py (hunk 2 above)
Comment (asadwaheed10) [reply tests/test_tempsensordb.py]: While debugging I was running into issues with this line, forgot to uncomment it out. I didn't bother changing anything to the test files as we won't be using those for the final demo. I can go back and fix this issue for the final commit as well.

File: tests/test_tempthingspeak.py (hunk 1 above)
Comment (asadwaheed10) [reply tests/test_tempthingspeak.py:9]: I don't think it's necessary, I only changed mine so there weren't any merge conflicts.

**Pull request merged**